processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1024,17 +1027,20 @@ public QueryableIndex load(File inDir, ObjectMapper mapper) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
-      Map<String, Object> metadata = null;
+      Metadata metadata = null;
       ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
       if (metadataBB != null) {
         try {
           metadata = mapper.readValue(
               serializerUtils.readBytes(metadataBB, metadataBB.remaining()),
-              new TypeReference<Map<String, Object>>()
-              {
-              }
+              Metadata.class
```

Comment:
for old segments metadata would just be null.
experimental simple kafka consumer firehose stored kafka offsets in metadata of persisted segments but did not include same in the handed off segment. theoretically, if someone used simple consumer kafka firehose and have intermediate persisted segments on realtime nodes then this new code will fail to deserialize that metadata and consider as if it wasn't there. But, noone but us used that so it is ok.